### PR TITLE
Further changes

### PR DESCRIPTION
**Pull request by MajorBreakfast**
- A missing word 😇
- `Future` -> `Async`
- Various improvements

[Rendered](https://github.com/MajorBreakfast/rfcs-1/blob/async-await/text/0000-async_await.md)

### CODE REVIEW

**Comment by MajorBreakfast** [comment]
Renamed `Item` to `Output` (mentioned by @alkis in https://github.com/rust-lang/rfcs/pull/2394#pullrequestreview-114971261)

**Comment by MajorBreakfast** [comment]
Don't merge yet, I want to add a commit that changes `Future` to `Async` Edit: Done

**Comment by MajorBreakfast** [comment]
- Renamed `Future` to `Async`
- Consistent highlighting for `async`, `await`

**Comment by MajorBreakfast** [comment]
Don't merge yet. I have [concerns](https://github.com/rust-lang/rfcs/pull/2418#issuecomment-384167584) that we'll be happy with our choice if we choose the name `Async`.

**Comment by MajorBreakfast** [comment]
As an experiment I've renamed `Async` to `Task`: [Rendered](https://github.com/MajorBreakfast/rfcs-1/blob/async-await/text/0000-async_await.md)

**Comment by MajorBreakfast** [comment]
My naming effort didn't catch on 😅`Async` it is

Here's the updated PR which renames everything from `Future` to `Async`

@aturon

**Comment by MajorBreakfast** [comment]
@aturon @withoutboats Any update on this?

File: text/0000-async_await.md [hunk 1]
```diff
@@ -69,14 +70,14 @@ async fn function(argument: &str) -> usize {
 }
 ```
 
-Async functions work differently from normal functions. When an async function
-is called, it does not enter the body immediately. Instead, it evaluates to an
-anonymous type which implements `Future`. As that future is polled, the
-function is evaluated up to the next `await` or return point inside of it (see
-the await syntax section next).
+`async` functions work differently than normal functions. When an `async`
```
Comment (withoutboats) [inline text/0000-async_await.md:73]: "different from" is correct: https://www.grammar.com/different-from-vs-different-than

File: text/0000-async_await.md [hunk 2]
```diff
@@ -284,28 +286,28 @@ async move {
...
+goals for 2018. `async` & `await` syntax based on the `Future` trait is the most
 expedient & low risk path to achieving that goal in the near future.
 
 This RFC, along with its companion lib RFC, makes a much firmer commitment to
-futures & async/await than we have previously as a project. If we decide to
+`Future` & `async`/`await` than we have previously as a project. If we decide to
 reverse course after stabilizing these features, it will be quite costly.
 Adding an alternative mechanism for asynchronous programming would be more
-costly because this exists. However, given our experience with futures, we are
+costly because this exists. However, given our experience with s, we are
```
Comment (withoutboats) [inline text/0000-async_await.md:304]: replaced "futures" with "s"

**Review by withoutboats — COMMENTED** [review]
In addition to the few specific comments, I think the introduction of code blocks here is a stylistic downgrade that makes the RFC less readable. Backticks should be used for keywords when discussing *the keyword* and for traits when discussing *the type*, not any time they are used. Several prominent examples:

* An async function is annotated with the `async` keyword, but that does not mean it should always be written "`async` function."
* A `return` expression must evaluate to the return type of the function, but that does not mean that type should be referred to as the "`return` type".
* A future is a type that implements `Future`, but that does not mean that you must always call them "`Future`s".
* You can "await" a future using the `await!` keyword, but that doesn't mean you must always "`await` a future" (indeed, in this case, in this RFC, any time await should be backticked it should be written `await!`, since it is a built-in macro and not normal syntax.)

Backticks should only be used when they improve understanding by clarifying that we are referring to a particular piece of code, not to the concept that that code would also refer to.

I'm going to merge the RFC now, but feel free to make another PR against the RFCs repo and request my review to make any of the other amendments this PR contains.

**Comment by MajorBreakfast** [comment]
Yes I wasn't so sure about whether styling all these words as code is a good idea myself. I'll create create a PR that only contains the other tweaks.

File: text/0000-async_await.md (hunk 1 above)
Comment (MajorBreakfast) [reply text/0000-async_await.md:73]: Interesting. Didn't know that. (I'm from Germany ;)

**Pull request closed without merging**